In [1]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

/kaggle/input/datasets-related-to-animation/Animation.csv
/kaggle/input/datasets-related-to-animation/Anime.csv


In [2]:
import seaborn as sns
import matplotlib.pyplot as plt

# **Anime Data Analytics**

In this python notebook, my aim is to analyze what is the top 10 best anime of all-time based on the dataset provided here. I have to be honest, as an anime fan, my favorite anime would be One Piece by Eiichiro Oda. I do hope that, the data shows the same here in this dataset. lol. But, we will be fair to the data and pronounce here, based on the dataset what the top 10 animes of all-time will be.

Let's begin the journey!
行くぞ!


# **Loading the Datasets**

Here I will load the dataset to see the structure of the data available

In [3]:
#anime data
anime_df = pd.read_csv('/kaggle/input/datasets-related-to-animation/Anime.csv')
anime_df.head(10)

animeID  name  studio  episodes  airing  rank  popularity  members  \
0        1     0       0        26   False    26          39   795733   
1        5     0       0         1   False   164         449   197791   
2        6     0       0        26   False   255         146   408548   
3        7     0       0        26   False  2371        1171    79397   
4        8     0       0        52   False  3544        3704    11708   
5       16     0       0        24   False   419         536   172274   
6       17     0       0        52   False  1186        3124    17145   
7       18     0       0        24   False   298        1262    73438   
8       19     0       0        74   False    49         156   394387   
9       20     0       0       220   False   705          10  1091313   

   favorites  reviewers  Animation_score  
0      43460     405664             8.81  
1        776     120243             8.41  
2      10432     212537             8.30  
3        537      32837             7.33  
4         14       4894             7.03  
5       3752      57065             8.12  
6        210       9307             7.67  
7        635      42555             8.23  
8      19188     131233             8.69  
9      39356     716412             7.90

In [4]:
#animation data.
animation_df = pd.read_csv('/kaggle/input/datasets-related-to-animation/Animation.csv')
animation_df.head(8)

animeID                             name    source             producers  \
0        1                     Cowboy Bebop  Original         Bandai Visual   
1        5  Cowboy Bebop: Tengoku no Tobira  Original               Sunrise   
2        6                           Trigun     Manga  Victor Entertainment   
3        7               Witch Hunter Robin  Original         Bandai Visual   
4        8                   Bouken Ou Beet     Manga              TV Tokyo   
5       16             Hachimitsu to Clover     Manga                 Genco   
6       17       Hungry Heart: Wild Striker     Manga                   NaN   
7       18           Initial D Fourth Stage     Manga           OB Planning   

           genre            studio  episodes  airing  rank  popularity  \
0         Action           Sunrise      26.0   False    26          39   
1         Action             Bones       1.0   False   164         449   
2         Action          Madhouse      26.0   False   255         146   
3         Action           Sunrise      26.0   False  2371        1171   
4      Adventure    Toei Animation      52.0   False  3544        3704   
5         Comedy         J.C.Staff      24.0   False   419         536   
6  Slice of Life  Nippon Animation      52.0   False  1186        3124   
7         Action          A.C.G.T.      24.0   False   298        1262   

   members  favorites  reviewers  Animation_score  
0   795733      43460     405664             8.81  
1   197791        776     120243             8.41  
2   408548      10432     212537             8.30  
3    79397        537      32837             7.33  
4    11708         14       4894             7.03  
5   172274       3752      57065             8.12  
6    17145        210       9307             7.67  
7    73438        635      42555             8.23

As seen in the datasets, the animation dataset is just the complete version of the anime dataset so I will just focus on the animation dataset. In this analysis I will try to answer based on the data, which is the all-time best anime among these list?

# **Data Cleaning and EDA**

Now, I will focus on cleaning the dataset and doing Exploratory Data Analysis with it after that.

First, we will check if there are missing data in the dataset.

In [5]:
animation_df.shape

(13631, 14)

This shows that there are 13,631 rows of data meaning there are 13,631 animes listed in this dataset and there are 14 columns which shows what kind of data was collected for this dataset.

Now I will check for missing datas in the dataset. First, I will check how many rows from the dataset has missing data.

In [6]:
na_rows = animation_df[animation_df.isna().any(axis=1)]
na_rows

animeID                        name       source producers  \
6           17  Hungry Heart: Wild Striker        Manga       NaN   
10          21                   One Piece        Manga   Fuji TV   
12          23            Ring ni Kakero 1        Manga       NaN   
15          27               Trinity Blood  Light novel       NaN   
76         106             Hana yori Dango        Manga       NaN   
...        ...                         ...          ...       ...   
13626    39189          Te Ke Dong Ji Pian      Unknown       NaN   
13627    39190          Wang Zhe Wai Chuan      Unknown       NaN   
13628    39191            Wan Jie Shen Zhu      Unknown       NaN   
13629    39192          Xia Gu Zhong An Zu      Unknown       NaN   
13630    39197      Karrimor Mountain Club     Original       NaN   

               genre            studio  episodes  airing   rank  popularity  \
6      Slice of Life  Nippon Animation      52.0   False   1186        3124   
10            Action    Toei Animation       NaN    True     94          36   
12            Action    Toei Animation      12.0   False   5641        6501   
15            Action             Gonzo      24.0   False   2081         695   
76             Drama    Toei Animation      51.0   False    819        2031   
...              ...               ...       ...     ...    ...         ...   
13626         Action               NaN      54.0   False  13301       15461   
13627         Comedy               NaN      12.0   False  13570       15472   
13628        Fantasy               NaN      60.0   False  13568       15470   
13629        Fantasy               NaN      12.0   False  13635       15473   
13630      Adventure               NaN       4.0   False  11624       15424   

       members  favorites  reviewers  Animation_score  
6        17145        210       9307             7.67  
10      803871      76869     465454             8.53  
12        2930         18        825             6.54  
15      137800       1403      61471             7.40  
76       37364        646      15258             7.84  
...        ...        ...        ...              ...  
13626        7          0          0              NaN  
13627        7          0          0              NaN  
13628        7          0          0              NaN  
13629        7          0          0              NaN  
13630       10          0          0              NaN  

[8713 rows x 14 columns]

As seen in the dataset, there are 8713 rows that have have missing data on them. The missing data comes mostly from the 'producers' columns and the 'studio' column which means the producers and the studio that made the animation are unknown. Also, the 'genre', 'episodes' and 'Animation_score' columns for some anime are also NA. To solve this, the producer and the studio columns will just be replaced as 'Unknown' since I don't need their information for the analysis. But for the other 3, I have to consider if I will use the data for determining the GOAT (Greatest of all-time) animes.

In [7]:
animation_df[['producers', 'studio']] = animation_df[['producers', 'studio']].fillna('Unknown')
animation_df

animeID                             name    source  \
0            1                     Cowboy Bebop  Original   
1            5  Cowboy Bebop: Tengoku no Tobira  Original   
2            6                           Trigun     Manga   
3            7               Witch Hunter Robin  Original   
4            8                   Bouken Ou Beet     Manga   
...        ...                              ...       ...   
13626    39189               Te Ke Dong Ji Pian   Unknown   
13627    39190               Wang Zhe Wai Chuan   Unknown   
13628    39191                 Wan Jie Shen Zhu   Unknown   
13629    39192               Xia Gu Zhong An Zu   Unknown   
13630    39197           Karrimor Mountain Club  Original   

                  producers      genre          studio  episodes  airing  \
0             Bandai Visual     Action         Sunrise      26.0   False   
1                   Sunrise     Action           Bones       1.0   False   
2      Victor Entertainment     Action        Madhouse      26.0   False   
3             Bandai Visual     Action         Sunrise      26.0   False   
4                  TV Tokyo  Adventure  Toei Animation      52.0   False   
...                     ...        ...             ...       ...     ...   
13626               Unknown     Action         Unknown      54.0   False   
13627               Unknown     Comedy         Unknown      12.0   False   
13628               Unknown    Fantasy         Unknown      60.0   False   
13629               Unknown    Fantasy         Unknown      12.0   False   
13630               Unknown  Adventure         Unknown       4.0   False   

        rank  popularity  members  favorites  reviewers  Animation_score  
0         26          39   795733      43460     405664             8.81  
1        164         449   197791        776     120243             8.41  
2        255         146   408548      10432     212537             8.30  
3       2371        1171    79397        537      32837             7.33  
4       3544        3704    11708         14       4894             7.03  
...      ...         ...      ...        ...        ...              ...  
13626  13301       15461        7          0          0              NaN  
13627  13570       15472        7          0          0              NaN  
13628  13568       15470        7          0          0              NaN  
13629  13635       15473        7          0          0              NaN  
13630  11624       15424       10          0          0              NaN  

[13631 rows x 14 columns]

In [8]:
na_rows = animation_df[animation_df.isna().any(axis=1)]
print(na_rows.shape)

print(animation_df.isna().sum())

(453, 14)
animeID              0
name                 0
source               0
producers            0
genre               62
studio               0
episodes           282
airing               0
rank                 0
popularity           0
members              0
favorites            0
reviewers            0
Animation_score    113
dtype: int64


Now, I only have 453 rows of data that have missing values in them. 